### ISSUE TITLE
Link on /marketing/connections needs discernible text

### ISSUE THREAD

**Issue by youbrokesomething** [Automattic/wp-calypso#75600]
### Quick summary

A link on /marketing/connections needs to have discernible text to give context for screen reader users.
The link is:
`<a class="inline-support-link" href="https://wordpress.com/support/post-automatically-to-social-media/" target="_blank" rel="noopener noreferrer">`

### Steps to reproduce

1. On /marketing/connections
2. Inspect the link above or use an accessibility checker like axe DevTools Chrome extension.

### What you expected to happen

All links should have discernible text to provide context for screen reader users so they know what the link is going to do.

### What actually happened

One of our help links doesn't have discernible text.

### Impact

Some (< 50%)

### Available workarounds?

No but the platform is still usable
